src/Core/ClassInfo.php

```diff
@@ -560,4 +561,51 @@ public static function parse_class_spec($classSpec)
         static::$_cache_parse[$classSpec] = $result;
         return $result;
     }
+
+
+    /**
+     * Returns a list of classes with a particular extension applied
+     *
+     * This reflects all extensions added (or removed) both via yaml as well as dynamically
+     * using Extensible::add_extension() and Extensible::removed_extension().
+     *
+     * Please not that
+     *
+     * @param string $extensionClass        extension class name
+     * @param string $baseClassOrObject     class or object to find subclasses of with the extension applied
+     * @param bool $includeBaseClass        include the base class itself?
+     * @param bool $includeSubclasses       include all levels of inheritance?
+     * @param bool $excludeTestOnly         exclude classes implementing TestOnly interface?
+     * @return array
+     * @throws \ReflectionException
+     */
+    public static function classesWithExtension(
+        string $extensionClass,
+        string $baseClassOrObject = DataObject::class,
+        bool $includeBaseClass = false,
+        bool $excludeTestOnly = true
```

Comment:
I'm pretty sure this is irrelevant. The `ClassManifest` (that powers `ClassInfo`) shouldn't even have `TestOnly` classes in environments where they're irrelevant. Considering this is code that exists only for testing purposes, I think it shouldn't be here.

Reply:
And how do you determine such environment? I think when they are in the code base, they are a part of the manifest.

Reply:
Interesting, I'd swear that TestOnly classes were appearing for me even outside phpunit runs, but they don't now... I'll remove this.